docs/roadmap.md

```diff
@@ -0,0 +1,81 @@
+# Project status & roadmap
+
+## Overview
+
+The Asterius project has come a long way and some examples with complex
+dependencies already work. It's still less mature than GHCJS yet; see the next
+section for details.
+
+In general, it's hard to give ETA for "production readiness", since improvements
+are continuous, and we haven't collected enough use cases from seed users yet.
+For more insight into what comes next for this project, we list our quarterly
+roadmap here.
+
+Besides the goals in each quarter, we also do regular maintainence like
+dependency upgrades and bugfixes. We also work on related projects (mainly
+`haskell-binaryen` and `inline-js`) to ensure they are kept in sync and also
+useful to regular Haskell developers.
+
+## What may stop one from using Asterius right now
+
+- Lack of JavaScriptCore/Safari support, due to incomplete JavaScript `BigInt`
+  support at the moment.
+- Runtime bugs. The generated code comes with a complex hand-written runtime
+  which is still buggy at times. The situation is expected to improve once we're
+  able to work with an IR more high-level than Cmm and shave off the current
+  hand-written garbage collector; see the 2020 Q3 section for explanation.
+- GHCJS projects aren't supported out of the box. Major incompatabilities
+  include:
+  - Word sizes differ. Asterius is still 64-bit based at the moment.
+  - JSFFI syntax and semantics differ. Asterius uses `Promise`-based async JSFFI
+    and GHCJS uses callbacks.
+  - Cabal handles GHCJS and Asterius differently.
+- Lack of Nix support.
+- Lack of GHCi support. TH support is not 100% complete; certain TH API which
+  requires preserving state across splices (e.g. `getQ`/`putQ`) don't work yet.
```

Comment:
I think it's best to split this into two separate bullets; even if the two issues might be related, they are separate issues.